src/typescript/frontend/src/lib/store/event-store.ts

```diff
@@ -305,8 +310,16 @@ export const createEventStore = (initialState: EventState = defaultState) => {
       // Note that we `unshift` here because we add the latest event to the front of the array.
       // We also update the latest bar if the incoming event is a swap or periodic state event.
       pushEventFromClient: (event) => {
-        if (get().guids.has(event.guid)) return;
-        addToLocalStorage(event);
+        if (get().guids.has(event.guid) && !event.guid.startsWith("Swap")) return;
+        if (
+          event.guid.startsWith("Swap") &&
+          (event as Types.SwapEvent).balanceAsFractionOfCirculatingSupply
```

Comment:
```suggestion
          typeof (event as Types.SwapEvent).balanceAsFractionOfCirculatingSupply === "bigint"
```

Zero is falsy here so if it's `0n` it won't enter the truthy half of the if/else block, gotta explicitly check for the type. You could also do `... !== "undefined"`